app/bot.go

```diff
@@ -4,6 +4,7 @@
 package app
 
 import (
+	"context"
```

Comment:
Do we have a future use for the context other than signaling where to run the query?
All this use of `context.Background()` feels a bit heavy on the code and possibly unnecessary in most cases.

Reply:
Indeed, it looks a bit heavy. But I think passing context is the right choice here. It's just that we aren't using context properly in our code. I would like to pass the context with a timeout to the actual DB query itself so that they can be properly cancelled if needed.

All store APIs should have passed a context anyways to start with. But it just looks like a lot of work because we never did that.

Thoughts?

Reply:
Sounds reasonable but the context should always come from the upper layer, correct? I see several additions in the store/search layer where a context is born out of nothing. So I am guessing it's really more of a `context.TODO()` for now.

Reply:
Fully agree. The context should come from the API layer, and _then_ passed on to the app layer. This would also help the open tracing layer which solves this by dynamically inserting contexts into the generated methods, but because of the fact that the contexts are not passed from the app layer to the store layer, we now have two different tracing layers, whereas ideally only a single context would need to be passed.

So yeah .. the context should come from upper layers. And I would very much like to see that in the near future :)